src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -61,8 +69,17 @@ public async Task Invoke(HttpContext context)
                 return;
             }
 
-            var authorizeResult = await policyEvaluator.AuthorizeAsync(policy, authenticateResult, context, resource: context);
-
+            object? resource;
+            if (_options.UseHttpContextAsResource)
+            {
+                resource = context;
+            }
+            else
+            {
+                resource = endpoint;
+            }
```

Comment:
```suggestion
            object? resource = _options.UseHttpContextAsResource ? context : endpoint;
```

Reply:
Ah right I tried this before it doesn't compile: Type of conditional expression cannot be determined because there is no implicit conversion between 'Microsoft.AspNetCore.Http.HttpContext' and 'Microsoft.AspNetCore.Http.Endpoint'